types/amazon-connect-streams/index.d.ts

```diff
@@ -0,0 +1,688 @@
+// Type definitions for non-npm package Amazon Connect Streams API 1.3
+// Project: amazon-connect-streams
+// Definitions by: Andy Hopper <https://github.com/andyhopp>
```

Comment:
There is already definition at https://github.com/aws/amazon-connect-streams/blob/master/src/index.d.ts
Since this is shipping with definition, I don't think you need them here.

Reply:
There isn't an NPM package for this library yet; I'm the author of that definition (used my work account) and have added it to DT in order to make it easier for developers

Reply:
Since the definitions are available where the source is, I don't think we need definitions on DT. https://github.com/aws/amazon-connect-streams#downloading-streams shows the steps to get the js files, .d.ts file should be available along with that?

Reply:
My chief concern is that, since we do not yet publish amazon-connect-streams as an NPM package, there are a couple of manual steps the developer would need to perform to modify their tsconfig to import the .d.ts file. Using the already-idiomatic @types scope simplifies the developer experience and is already a familiar process for developers using external JS libraries.
 
Once we start publishing ACS to NPM, we would, of course, remove the DT package.
 
Thanks again!

Reply:
Adding @RyanCavanaugh and @sandersn for their opinion on hosting the types while they are still available elsewhere.

Reply:
I should add; if having the types in the ACS GH repo is the issue, I could certainly remove them (I'd prefer *not* to, of course, but I'm willing to be persuaded otherwise!). I'm just trying to provide developers with a simple/easy way to code against ACS with Typescript while I drum up support for an official NPM package.

Reply:
@andyhop making typings available easily seems to make sense. We see that your steps for install are to clone the repro, build and then copy js. Why not copy `.d.ts` as well next to the .js file? When the package is ready to ship, you can add `typings` field to package json pointing to the `.d.ts`

Reply:
My concern there is that this becomes an additional step the developer needs to perform (modifying the tsconfig file to add the types). Most TS developers are familiar with the default behavior (to wit: including node_modules/@types), but once you start hacking the types property the developer must explicitly include that as well as the other types.

I certainly could work around this by instructing the developer to manually create a @types/amazon-connect-streams sub-folder, but I feel like that's an abuse.

It sounds like we're over-indexing on the fact that I included the types in the JS repo (which is officially only the JS library). I'll work with that team to address the fact that it's not published to NPM, but in the meantime, I really would like to keep the number of manual steps they need to perform to a minimum. I'm currently unhappy with the fact that users have to run a makefile to create the .js file for the library (again, working with the team to resolve that), and I'd be even unhappier with them having to know how to tweak up their tsconfig file (speaking from experience: the less you touch that file, the better).

Would deleting the index.d.ts file from the amazon-connect-streams repo resolve this to everyone's satisfaction?

Reply:
I think the main concern is that we don't understand the extra step you are talking about. Why is copying the .d.ts file next to `.js` file while you are copying it difficult compared to installing types package?

Reply:
Because if the developer is working with something like React or Angular, the location of the external javascript libraries is not typically under the project folder created by "ng new" or "create-react-app" and will therefore require modifying the tsconfig file to pull that file in. Performing an "npm i @types/xxx" is something I can expect most TS developers to be familiar with.